### PR TITLE
Add support for load multiple files by values defined in NODE_ENV

### PR DESCRIPTION
**Pull request by leonardovillela**
See #481, @lorenwest.

### CODE REVIEW

**Comment by leonardovillela** [comment]
Hello Guys, any problem with this PR? I can help with something?

**Comment by lorenwest** [comment]
Thank you for your patience - I've been pretty busy lately.

File: test/14-config/bare-metal.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "db": {
+    "name": "bare-metal-config-env-provided"
+  }
+}
```
Comment (lorenwest) [inline test/14-config/bare-metal.json]: Looks like you've removed the final newline in these files. Could you un-do those changes?

**Comment by lorenwest** [comment]
This looks good - thank you for adding this feature. Only one comment in the code review - if you could undo those newline removals and re-issue the PR, I'll get it merged.

File: test/14-config/bare-metal.json (hunk 1 above)
Comment (leonardovillela) [reply test/14-config/bare-metal.json]: Yes, i will update this

**Comment by lorenwest** [comment]
@leonardovillela - We are having a conversation on #484 regarding including this in a major version bump. Will get this published as soon as we conclude that version bump issue.

**Comment by KrishnaPG** [comment]
Would `default` be still loaded in case multiple env are specified?

**Comment by markstos** [comment]
I noticed the wiki was never updated to reflect this change. Documenting it now.

**Pull request merged**